tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -472,14 +472,15 @@ def read_model(path_to_xml):
         t = tm.Telemetry()
         t.send_event('mo', 'offline_transformations_status', message)
 
-        if return_code != 0:
-            raise Error("offline transformations step has failed.")
-
         for suf in [".xml", ".bin", ".mapping"]:
             # remove existing files
             path_to_file = orig_model_name + "_tmp" + suf
             if os.path.exists(path_to_file):
                 os.remove(path_to_file)
+
+        if return_code != 0:
+            raise Error("offline transformations step has failed.")
+
     return func
 
```

Comment:
what if we have an exception in IR Frontend during model reading. Again the exception is not handled and the tmp IR will not be deleted.
Also the exception can be generated in any place from the moment of its serialization, not only in IR Frontend.

Reply:
I understand what you mean but your question relates to a different topic: how to handle all exceptions during conversion given the fact that they can be raised in a very different places and conditions.

Subject of this PR was to remove `_tmp.bin, _tmp.xml, _tmp.mapping` regardless of whether exception happened or not tmp files must be cleaned. Previously raising of an Error was done before cleaning tmp files therefore these leftover tmp files led to confusion during accuracy check. All i did here is just moved raising Error after cleaning up tmp files.

And about testing. I don't thing we have here anything to test. We just clean these tmp files and that't all. Considering the information above do you still want me to write test?

Reply:
@pavel-esir, please address the remained concern in the separate PR

Reply:
addressed comments in this PR https://github.com/openvinotoolkit/openvino/pull/14712